client/components/BudgetGrid/EntryFormRow.js

```diff
@@ -0,0 +1,66 @@
+import React, { Component, PropTypes } from 'react';
+
+class EntryFormRow extends Component {
+  static propTypes = {
+    categories: PropTypes.object.isRequired
+  }
+
+  state = {
+    categoryId: 16,
+    description: '',
+    amount: ''
+  }
+
+  getCategoryOptions() {
```

Comment:
instead of using factory methods, let's create a functional component (it can be inside this file/module too) that returns the entire `<select>` with `<options>`. 

Functional components > factory